gradle-jdks-settings/src/main/java/com/palantir/gradle/jdks/settings/CurrentOs.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks.settings;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Simplified version of CurrentOs.java.
+ * @see <a href="file:../gradle-jdks-setup-common/src/main/java/com/palantir/gradle/jdks/CurrentOs.java>CurrentOs.java</a>
+ * We cannot depend directly on `gradle-jdks-setup-common` as it might lead to Gradle classLoader issues.
+ */
```

Comment:
Grim these four files need copying. I think we might be able to shade the dep they come from instead to avoid the copying using gradle-shadow-jar?